components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -159,6 +161,39 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
             return Task.CompletedTask;
         }
 
+        /// <inheritdoc/>
+        public Task<JobDiagnosticInfoModel> GetDiagnosticInfo() {
+            var totalSeconds = (DateTime.UtcNow - _diagnosticStart).TotalSeconds;
+            double totalDuration = _diagnosticStart != DateTime.MinValue ? totalSeconds : 0;
+            double chunkSizeAverage = _messageEncoder.AvgMessageSize / (4 * 1024);
+            double sentMessagesPerSec = totalDuration > 0 ? _messageSink.SentMessagesCount / totalDuration : 0;
+            double estimatedMsgChunksPerDay = Math.Ceiling(chunkSizeAverage) * sentMessagesPerSec * 60 * 60 * 24;
+
+            _diagnosticInfo.Id = Name;
+            _diagnosticInfo.IngestionDuration = TimeSpan.FromSeconds(totalDuration);
+            _diagnosticInfo.IngressDataChanges = _messageTrigger.DataChangesCount;
+            _diagnosticInfo.IngressValueChanges = _messageTrigger.ValueChangesCount;
+            _diagnosticInfo.IngressBatchBlockBufferSize = _batchDataSetMessageBlock.OutputCount;
+            _diagnosticInfo.EncodingBlockInputOutputSize = _messageEncoder.NotificationsProcessedCount;
+            _diagnosticInfo.EncoderNotificationsProcessed = _messageEncoder.NotificationsProcessedCount;
+            _diagnosticInfo.EncoderNotificationsDropped = _messageEncoder.NotificationsDroppedCount;
+            _diagnosticInfo.EncoderIoTMessagesProcessed = _messageEncoder.MessagesProcessedCount;
+            _diagnosticInfo.EncoderAvgNotificationsMessage = _messageEncoder.AvgNotificationsPerMessage;
+            _diagnosticInfo.EncoderAvgIoTMessageBodySize = _messageEncoder.AvgMessageSize;
+            _diagnosticInfo.EncoderAvgIoTChunkUsage = chunkSizeAverage;
+            _diagnosticInfo.EstimatedIoTChunksPerDay = estimatedMsgChunksPerDay;
+            _diagnosticInfo.OutgressBatchBlockBufferSize = _batchNetworkMessageBlock.OutputCount;
+            _diagnosticInfo.OutgressInputBufferCount = _sinkBlock.InputCount;
+            _diagnosticInfo.OutgressInputBufferDropped = _sinkBlockInputDroppedCount;
+            _diagnosticInfo.OutgressIoTMessageCount = _messageSink.SentMessagesCount;
+            _diagnosticInfo.ConnectionRetries = _messageTrigger.NumberOfConnectionRetries;
+            _diagnosticInfo.OpcEndpointConnected = _messageTrigger.IsConnectionOk;
+            _diagnosticInfo.MonitoredOpcNodesSucceededCount = _messageTrigger.NumberOfGoodNodes;
+            _diagnosticInfo.MonitoredOpcNodesFailedCount = _messageTrigger.NumberOfBadNodes;
+
+            return Task.FromResult<JobDiagnosticInfoModel>(_diagnosticInfo);
```

Comment:
Seems like we are constructing the whole `JobDiagnosticInfoModel` object above, so maybe we can get rid of `_diagnosticInfo` member variable ?

Reply:
_diagnosticInfo not needed, removed